socorro/tests/external/es/test_supersearch.py

```diff
@@ -500,35 +494,46 @@ def test_get_with_combined_operators(self, es_helper):
         es_helper.index_crash(
             processed_crash={
                 "uuid": create_new_ooid(timestamp=now),
-                "signature": sigs[3],
+                "signature": "foo(bar)",
                 "app_notes": "mozilla bar",
                 "product": "EarthRacoon",
                 "date_processed": now,
             },
         )
         es_helper.refresh()
 
+        # Search for signatures matching "js" or containing "::"
         res = api.get(signature=["js", "~::"])
         assert res["total"] == 3
-        assert sorted([x["signature"] for x in res["hits"]]), sorted(
-            [sigs[0], sigs[1], sigs[2]]
-        )
+        signatures = [x["signature"] for x in res["hits"]]
+        assert list(sorted(signatures)) == [
+            "js::break_your_browser",
+            "js<isKewl>",
+            "mozilla::js::function",
+        ]
 
+        # Search for signatures matching "js" or containing "::" AND product is
+        # "Unknown" -- there are none
         res = api.get(signature=["js", "~::"], product=["Unknown"])
         assert res["total"] == 0
         assert len(res["hits"]) == 0
 
-        res = api.get(signature=["js", "~::"], product=["WaterWolf", "EarthRacoon"])
-        assert res["total"] == 3
-        assert sorted([x["signature"] for x in res["hits"]]) == sorted(
-            [sigs[0], sigs[1], sigs[2]]
-        )
-
-        res = api.get(signature=["js", "~::"], app_notes=["foo bar"])
```

Comment:
This is the case that fails with the new code. What used to happen is it did a "terms" query and only returned the item that matched exactly "foo bar". But that's not what the default operator should be doing--that's what the "is exactly" operator does.

It started failing because the search text gets analyzed per the field `storage_mapping` so this gets tokenized into "foo" and "bar" and those strings appear in several crash report data app_notes values.